emcc.py

```diff
@@ -1431,6 +1445,10 @@ def check(input_file):
       options.separate_asm = True
       shared.Settings.FINALIZE_ASM_JS = False
 
+    # MINIMAL_RUNTIME always use separate .asm.js file for best performance and memory usage
+    if shared.Settings.MINIMAL_RUNTIME and not shared.Settings.WASM:
+      options.separate_asm = True
+
```

Comment:
Always use separate .asm.js files, so that it can be asynchronously loaded, debugging the main JS is feasible when the compiled asm.js code is out of sight, and Firefox can unload asm.js script content from memory to disk.